src/components/page-deliveries/page-deliveries.tsx

```diff
@@ -383,18 +395,14 @@ export class PageDeliveries {
               <h3>Previous Deliveries</h3>
               <ul>
                 {selectedLocation != null ? (
-                  <OrderAndTruckInfoList
-                    items={previousItems.slice(0, 3)}
-                    selectedLocation={selectedLocation}
-                    onOrderSelected={order => (this.selectedOrder = this.recentOrders?.find(x => x.id === order.id))}
-                  />
+                  <OrderAndTruckInfoList items={previousItems.slice(0, 3)} selectedLocation={selectedLocation} onOrderSelected={order => (this.selectedOrder = order)} />
                 ) : (
-                  (items?.length > 3 ? items?.slice(3, 6) : ([null, null, null] as (OrderOrTruckItem | null)[])) // show placeholders if no data
+                  (combinedItems?.length > 0 ? combinedItems?.slice(0, 6) : ([null, null, null] as (OrderOrTruckItem | null)[])) // show placeholders if no data
                     .map(x =>
                       x != null && x.type === "truck" ? (
                         <TruckInfoDisplay truck={x.data} onClick={() => this.selectLocation(x?.data?.location)} noIcon={true} />
                       ) : (
-                        <OrderDetailDisplay order={x?.data} onClick={() => this.selectLocation(x?.data?.location)} noIcon={true} />
+                        <OrderDetailsDisplay reportCount={x?.data?.reports?.length || 0} order={x?.data} onClick={() => this.selectLocation(x?.data?.location)} noIcon={true} />
```

Comment:
`x?.data?.reports?.length || 0` and `x?.data?.reports?.length ?? 0` are subtly different. I doubt that difference matters here, but as a matter of habit, I prefer `??`. See [Nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_Coalescing_Operator).